neo4j/src/test/java/org/hibernate/ogm/datastore/neo4j/test/embedded/dialect/EmbeddedNeo4jBackendQueryResultIteratorTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.ogm.datastore.neo4j.test.embedded.dialect;
+
+import org.hibernate.ogm.datastore.neo4j.embedded.dialect.impl.EmbeddedNeo4jBackendQueryResultIterator;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.junit.Test;
+import org.neo4j.graphdb.Result;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+
+public class EmbeddedNeo4jBackendQueryResultIteratorTest {
```

Comment:
Maybe we should add a JavaDoc here, saying that we're testing the order consistency?

Reply:
Thanks a lot